totalRP3/modules/register/characters/PlayerMapScan/PlayerMapPin.lua

```diff
@@ -0,0 +1,121 @@
+---    See the License for the specific language governing permissions and
+---    limitations under the License.
+----------------------------------------------------------------------------------
+
+---@type TRP3_API
+local _, TRP3_API = ...;
+local Ellyb = TRP3_API.Ellyb;
+---@type AddOn_TotalRP3
+local AddOn_TotalRP3 = AddOn_TotalRP3;
+
+--region Lua imports
+local huge = math.huge;
+--endregion
+
+--region Total RP 3 imports
+local loc = TRP3_API.loc;
+--endregion
+
+--region Ellyb imports
+local ORANGE = Ellyb.ColorManager.ORANGE;
+---endregion
+
+-- Create the pin template, above group members
+TRP3_PlayerMapPinMixin = AddOn_TotalRP3.MapPoiMixins.createPinTemplate(
+	AddOn_TotalRP3.MapPoiMixins.GroupedCoalescedMapPinMixin, -- Use coalesced grouped tooltips (show multiple player names)
+	AddOn_TotalRP3.MapPoiMixins.AnimatedPinMixin -- Use animated icons (bounce in)
+);
+
+-- Expose template name, so the scan can use it for the MapDataProvider
+TRP3_PlayerMapPinMixin.TEMPLATE_NAME = "TRP3_PlayerMapPinTemplate";
+
+--- This is called when the data provider acquire a pin, to transform poiInfo received from the scan
+--- into display info to be used to decorate the pin.
+function TRP3_PlayerMapPinMixin:GetDisplayDataFromPoiInfo(poiInfo)
+	local characterID = poiInfo.sender;
+	local displayData = {};
+	local isSelf = characterID == TRP3_API.globals.player_id;
+
+	if not isSelf and (not TRP3_API.register.isUnitIDKnown(characterID) or not TRP3_API.register.hasProfile(characterID)) then
+		-- Only remove the server name from the sender ID
+		displayData.playerName = characterID:gsub("%-.*$", "");
+		return displayData;
+	end
+
+	local profile;
+	if isSelf then
+		-- Special case when seeing ourselves on the map (DEBUG)
+		profile = TRP3_API.profile.getPlayerCurrentProfile().player;
+		displayData.iconAtlas = "PlayerPartyBlip";
+		displayData.iconColor = Ellyb.ColorManager.CYAN;
+		displayData.categoryName = loc.REG_RELATION .. ": " .. Ellyb.ColorManager.CYAN("SELF");
+		displayData.categoryPriority = -huge;
+	else
+		profile = TRP3_API.register.getUnitIDCurrentProfile(characterID);
+
+		--region Player relationship
+		local relation = TRP3_API.register.relation.getRelation(TRP3_API.register.getUnitIDProfileID(characterID));
+		if relation ~= TRP3_API.register.relation.NONE then
+			local relationShipColor = TRP3_API.register.relation.getColor(relation);
+			-- Swap out the atlas for this marker.
+			displayData.iconAtlas = "PlayerPartyBlip";
+			displayData.iconColor = relationShipColor;
+
+			-- Store the relationship on the marker itself as the category.
+			displayData.categoryName = loc.REG_RELATION .. ": " .. relationShipColor(loc:GetText("REG_RELATION_".. relation));
+			displayData.categoryPriority = TRP3_API.globals.RELATIONS_ORDER[relation] or -huge;
+		end
+		--endregion
+	end
+
+	--region Player name
+	displayData.playerName = TRP3_API.register.getCompleteName(profile.characteristics, characterID, true);
```

Comment:
You shouldn't send characterID without removing realm from it. If profile.characteristics or profile.characteristics.FN is nil, you'll have the realm in displayData.playerName